lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -319,14 +324,28 @@ def _format_datetime(dt):
         """
         return dt.isoformat().replace('+00:00', 'Z')
 
+    def get_posting_status(posting_restrictions, blackout_schedules):
+        now = datetime.now(UTC)
+        if posting_restrictions == PostingRestriction.Disabled:
+            return True
+        elif posting_restrictions == PostingRestriction.Scheduled:
+            return not any(schedule["start"] <= now <= schedule["end"] for schedule in blackout_schedules)
+        else:
+            return False
```

Comment:
This function is not returning posting status... Better name would be

`def is_posting_allowed(posting_restrictions, blackout_schedules):`

`is_posting_enabled = is_posting_allowed(course_config.posting_restrictions, course.get_discussion_blackout_datetimes())`